File: src/main/scala/algo/InsertionSort.scala
package algo

import scala.collection.mutable.ArrayBuffer

object InsertionSort extends App {
  var arr = ArrayBuffer(20, 35, -15, 7, 55, 1, -22, 34, 67, -2, 14, 100, -3, -9)
  for(i <- 1 until arr.length){
    val temp = arr(i)
    var j = i-1
    var k = i
    while(j >= 0 && arr(j)> temp){
      arr(k) = arr(j)
      arr(j) = temp
      j= j-1
      k = k -1
    }

  }
  arr.foreach(println)
}

File: src/main/java/InsertionSort.java
public class InsertionSort {
    public static void main(String[] args) {
        int[] arr = {200, 35, -15, 7, 55, 1, -22, 34, 67, -2, 14, 100, -3, -9};
        for (int i = 1; i < arr.length; i++) {
            int j = i-1;
            int temp = arr[i];
            while(j>= 0 && arr[j]>temp){
                arr[i] = arr[j];
                arr[j] = temp;
                j--;
                i--;
            }
        }

        for(int k : arr){
            System.out.print(k+" ");
        }

    }
}
